db/repair_test.cc

```diff
@@ -174,6 +174,38 @@ TEST_F(RepairTest, UnflushedSst) {
   ASSERT_EQ(Get("key"), "val");
 }
 
+TEST_F(RepairTest, SeparateWalDir) {
+  do {
+	Options options = CurrentOptions();
+	DestroyAndReopen(options);
+	Put("key", "val");
+	Put("foo", "bar");
+	VectorLogPtr wal_files;
+	ASSERT_OK(dbfull()->GetSortedWalFiles(wal_files));
+	ASSERT_EQ(wal_files.size(), 1);
+	uint64_t total_ssts_size;
+	GetAllSSTFiles(&total_ssts_size);
+	ASSERT_EQ(total_ssts_size, 0);
+	std::string manifest_path =
+		DescriptorFileName(dbname_, dbfull()->TEST_Current_Manifest_FileNo());
+
+	Close();
+	ASSERT_OK(env_->FileExists(manifest_path));
+	ASSERT_OK(env_->DeleteFile(manifest_path));
+
+	ASSERT_OK(RepairDB(dbname_, options));
+
+	Reopen(options);
+
+	ASSERT_OK(dbfull()->GetSortedWalFiles(wal_files));
+	ASSERT_EQ(wal_files.size(), 0);
+	GetAllSSTFiles(&total_ssts_size);
+	ASSERT_GT(total_ssts_size, 0);
```

Comment:
Can you move this check before `Reopen()`? Otherwise the test passes without your code change. Idk why, but our DB opening code processes all logs in `wal_dir`, even if they're not in the MANIFEST file: https://github.com/facebook/rocksdb/blob/master/db/db_impl_open.cc#L370.